packages/iTwinUI-react/src/core/Table/columns/actionColumn.tsx

```diff
@@ -36,6 +36,8 @@ const ACTION_CELL_ID = 'iui-table-action';
  */
 export const ActionColumn = <T extends Record<string, unknown>>({
   columnManager = false,
+  className = '',
+  style = {},
 } = {}) => {
```

Comment:
I don't think these props make sense as top level arguments of `ActionColumn` because the dropdown is only a small part of the column.

Also, instead of plucking individual props we should probably expose either the whole dropdown component or all of its props.

I'm thinking we could convert `columnManager` into an object that has a property called `dropdownMenuProps`. The end usage could look something like:
```ts
ActionColumn({
  columnManager: {
    dropdownMenuProps: { className: 'my-custom-class' }
  }
})
```

Here's how you would change the type to be polymorphic (accept either a boolean or an object):
```suggestion
export const ActionColumn = <T extends Record<string, unknown>>({
  columnManager = false,
}: {
  columnManager?: boolean | { dropdownMenuProps: PopoverProps }
} = {}) => {
```

Reply:
Thanks, yes this is a great idea. Just to confirm, are you meaning that I should also remove `className`? Or should I keep className as one of the props but also add `columnManager` as an object of type `PopoverProps`? I wanted to confirm because `PopoverProps` did not have `className` as a property.

Reply:
> `PopoverProps` did not have `className` as a property.

oh then you can use `DropdownMenuProps` but might need to `Omit` some props from it. We definitely want to keep `className` as that's the main reason for this change.

Reply:
I actually just realized that I might not have understood the way these `type`s work. Looks like the hierarchy for `PopoverProps` is `PopoverProps -> TippyProps -> className`. `DropdownMenu` uses `PopoverProps` and also uses `CommonProps -> StylingProps -> ClassNameProps -> className`. But I'm not sure if this is correct.

```tsx
export type DropdownMenuProps = {...} & PopoverProps & Omit<CommonProps, 'title'>;
export type PopoverProps = {...} & Omit<TippyProps, 'placement' | 'trigger' | 'visible'>;
export type CommonProps = {...} & StylingProps;
export type StylingProps = {...} & ClassNameProps;

export type ClassNameProps = { className?: string; };
export interface TippyProps extends Partial<Omit<Props, 'content' | 'render'>> {
  ...
  className?: string;
  render?: (
    ...
  ) => React.ReactNode;
}


```

What I ended up doing was to create a new type called `ActionColumnDropdownMenuProps` that extended `DropdownMenuProps`. I omitted the three properties that were being passed by the ActionColumn to the `DowndownMenu`.

```tsx
export type ActionColumnDropdownMenuProps = Omit<
  DropdownMenuProps,
  'menuItems' | 'onHide' | 'onShow'
>;
```

I just pushed my changes. Cann you please confirm if I did everything correctly like you suggested?

Reply:
that looks fine. two minor suggestions:
- I would also probably omit `children`
- no need to `export` this type, it can stay internal

Reply:
**[DONE]** Thanks, I made these changes. But just curious for my understanding, what does the `export` do here? What are the benefits of not exporting it here? Would it affect the autocomplete of the available parameters for the `ActionColumnDropdownMenuProps` in any way?

Reply:
`export` makes it available outside the file (so it can be `import`ed in another file). not exporting allows us to make changes to it in the future without potentially breaking anything on the outside.